src/main/java/mate/academy/spring/config/SecurityConfig.java

```diff
@@ -1,27 +1,49 @@
 package mate.academy.spring.config;
 
+import mate.academy.spring.model.Role;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.context.annotation.Bean;
+import org.springframework.http.HttpMethod;
 import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
-import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.core.userdetails.UserDetailsService;
 import org.springframework.security.crypto.password.PasswordEncoder;
 
 @EnableWebSecurity
 public class SecurityConfig extends WebSecurityConfigurerAdapter {
+    private final UserDetailsService userDetailsService;
+    private final PasswordEncoder passwordEncoder;
+
+    public SecurityConfig(UserDetailsService userDetailsService, PasswordEncoder passwordEncoder) {
+        this.userDetailsService = userDetailsService;
+        this.passwordEncoder = passwordEncoder;
+    }
+
     @Autowired
     public void configure(AuthenticationManagerBuilder auth) throws Exception {
-        auth
-                .inMemoryAuthentication()
-                .passwordEncoder(getEncoder())
-                .withUser("1@123").password(getEncoder().encode("123")).roles("USER");
+        auth.userDetailsService(userDetailsService).passwordEncoder(passwordEncoder);
     }
 
     protected void configure(HttpSecurity http) throws Exception {
         http
                 .authorizeRequests()
+                .antMatchers("/users/by-email").hasRole(Role.RoleName.ADMIN.name())
+                .antMatchers("/shopping-carts/**").hasRole(Role.RoleName.USER.name())
+                .antMatchers("/orders/**").hasRole(Role.RoleName.USER.name())
+                .antMatchers(HttpMethod.GET, "/movie-sessions/**").permitAll()
+                .antMatchers(HttpMethod.POST, "/movie-sessions/**")
+                    .hasRole(Role.RoleName.ADMIN.name())
+                .antMatchers(HttpMethod.PUT, "/movie-sessions/**")
+                    .hasRole(Role.RoleName.ADMIN.name())
+                .antMatchers(HttpMethod.DELETE, "/movie-sessions/**")
+                    .hasRole(Role.RoleName.ADMIN.name())
+                .antMatchers(HttpMethod.GET, "/movies/**").permitAll()
+                .antMatchers(HttpMethod.POST, "/movies/**").hasRole(Role.RoleName.ADMIN.name())
+                .antMatchers(HttpMethod.GET, "/cinema-halls/**").permitAll()
+                .antMatchers(HttpMethod.POST, "/cinema-halls/**")
+                    .hasRole(Role.RoleName.ADMIN.name())
+                .antMatchers("/register").hasRole(Role.RoleName.ADMIN.name())
```

Comment:
Role.RoleName.ADMIN.name() and Role.RoleName.USER.name() used so much so it's worth to extract them to variables or make static imports to use ADMIN.name() and USER.name() instead of full form